EZEE-3337 - Image Editor configuration (#1265)
ezsystems/developer-documentation: docs/guide/image_editor.md

> # Configuring the Image Editor

Reviewer: Please add an Experience/Commerce badge, as https://github.com/ezsystems/developer-documentation/pull/1230 has been merged and image editor is in those two editions only.

> ## Introduction
> 

Reviewer: Not neded
```suggestion
```

> When a Content Item contains Fields of the [ezimage](../api/field_type_reference.md#image-field-type) or [ezimageasset](../api/field_type_reference.md#imageasset-field-type) type, users can perform basic image editing functions with the Image Editor.

Reviewer: ```suggestion
When a Content item contains Fields of the [ezimage](../api/field_type_reference.md#image-field-type) or [ezimageasset](../api/field_type_reference.md#imageasset-field-type) type, users can perform basic image editing functions with the Image Editor.
```

Reviewer 2: I think it is worth mentioning that its not working on DAM assets atm (button would be disabled/invisible).

For more information, see the [user documentation](https://doc.ibexa.co/projects/userguide/en/master/editing_images/).

## Configuration

> You can modify the default settings to change the appearance or behaviour of the Image Editor.

Reviewer: AmE
```suggestion
You can modify the default settings to change the appearance or behavior of the Image Editor.
```

You can also expand the default set of parameters to create buttons that may be required by custom features that you add by [extending the Image Editor](#extending-the-image-editor).

> To do this, modify the `config/packages/ezplatform.yaml` file, or create a separate YAML file in the `config/packages` folder, and add a settings tree similar to the following example.

Reviewer: ```suggestion

To do this, modify the `config/packages/ezplatform.yaml` file, or create a separate YAML file in the `config/packages` folder, and add a settings tree similar to the following example.
```

> Image Editor settings are [dynamic](config_dynamic.md), therefore you can configure a different set of buttons or preset values for each of the SiteAccesses in your installation.

Reviewer: We use "SiteAccess-aware" throughout the rest of the doc, should be enough here:
```suggestion
Image Editor settings are SiteAccess-aware.
```

> The example sets the aspect ratio values and label names for buttons used by the Crop feature.

Reviewer: ```suggestion

The following example sets the aspect ratio values and label names for buttons used by the Crop feature.
```

> 

Reviewer: ```suggestion
```

``` yaml
system:

>     <siteaccess_name>:

Reviewer: ```suggestion
    <siteaccess>:
```

        image_editor:
            action_groups:
                default:

>                     id: default

Reviewer 3: I would mention that developers can create more than one group.

                    label: Default
                    actions:
                        (...)

>                         crop:

Reviewer 3: I would mention that each action can have `priority` and `visible` - thanks to that you can change the order of actions and remove some actions from the image editor.

                            id: crop
                            buttons:
                                1-1:
                                    label: 1:1
                                    ratio:
                                        x: 1
                                        y: 1
                                3-4:
                                    label: 3:4
                                    ratio:
                                        x: 3
                                        y: 4
                                4-3:
                                    label: 4:3
                                    ratio:
                                        x: 4
                                        y: 3
                                16-9:
                                    label: 16:9
                                    ratio:
                                        x: 16
                                        y: 9
                                custom:
                                    label: Custom
```

### Additional information

Each image can be accompanied by additional information that is not visible to the user.

> By default, additional information serves to store the coordinates of the [focal point](https://doc.ibexa.co/projects/userguide/en/master/editing_images/#focal-point), but you can use this extension point to pass various parameters of custom features that you add by [extending the Image Editor](#extending-the-image-editor).

Reviewer: ```suggestion
By default, additional information stores the coordinates of the [focal point](https://doc.ibexa.co/projects/userguide/en/master/editing_images/#focal-point),
but you can use this extension point to pass various parameters of custom features
that you add by [extending the Image Editor](#extending-the-image-editor).
```

To modify the value of additional information programmatically, you can set a value of the `Image` field by using the PHP API, for example:

``` php
new FieldValue([
     'data' => [
         'width' => '100',
         'height' => '200',
         'alternativeText' => 'test',
         'mime' => 'image/png',
         'id' => 1,
         'fileName' => 'image.png',
         'additionalData' => [
             'focalPointX' => 50,
             'focalPointY' => 100,
             'author' => 'John Smith',
         ],
     ],
 ]),
```
 

> ## Extending the Image Editor

Reviewer: Not sure if it makes sense to include this paragraph, if we're not describing the procedure in detail.
If this is is something we should have a detailed guide on, let's add is as a follow-up.
Then also remove the link above.

You can extend the Image Editor if your users require additional features.

> For example, you can create a feature that modifies the colour palette of an image.

Reviewer: ```suggestion
For example, you can create a feature that modifies the color palette of an image.
```
